### PR TITLE
OCEANWATER-1207 Plans that clear goal errors

### PR DESCRIPTION
**Pull request by kmdalal**
### Summary
- New plan `ClearGoalErrors` that continuously monitors for goal errors and clears them upon occurrence.
- Use of `ClearGoalErrors` in reference mission plans.
- Minor unrelated edits.

### Test

The `ow_simulator` must be on the epic branch `OCEANWATER-1205_blocking_faults`

1. Start the Atacama world and run the `ClearGoalErrors` plan.
2. Induce pan/tilt, arm, and task goal errors in any order and watch them get noticed and then cleared.  See the plan `FaultClearDemo` for ways to induce goal errors in these subsystems.  Another way to induce a Task goal error is invoking `arm_stop` during any `Task*` action.
3. When satisfied, Ctrl-C to quit the exec node (but not the simulator).
4. Regression test: restart the exec launch and run 'ReferenceMission1', which now monitors for goal errors using `ClearGoalErrors`.  There should be none, though you should see this message near the start of the scenario: `PLEXIL: ClearGoalErrors is running. Goal errors will automatically be cleared.`

### Caveat
Due to a known bug, camera goal errors can be induced via publishing a camera trigger message (e.g. through RQT) but not the `camera_capture.py` script.

### Caveat
Due to a known bug, camera goal errors can be induced via publishing a camera trigger message (e.g. through RQT) but not the `camera_capture.py` script.

### CODE REVIEW

File: ow_plexil/src/plans/ClearGoalErrorAttempt.plp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+// Reports and attempts to clears the given goal error. Verification
+// needs to be done in the calling plan, which was intended to be
+// ClearGoalErrors.plp.
```
Comment (AstroStucky) [inline ow_plexil/src/plans/ClearGoalErrorAttempt.plp:7]:
```suggestion
// needs to be done in the calling plan.
// Designed as a sub-plan of ClearGoalErrors.plp.
```
This statement was clear and I think it's helpful here too.

File: ow_plexil/src/plans/FaultClearDemo.plp [hunk 2]
```diff
@@ -16,9 +16,8 @@
 
 // To test the antenna, try 'rosrun ow_lander pan.py 100'.
 
-// Note that goal errors are associated only with ROS actions, and not
-// other means of controlling lander components such as ad hoc message
-// publishing.
+// Note that goal errors are triggered only by ROS actions, and not by
+// other means of lander control such as ad hoc message publishing.
```
Comment (AstroStucky) [inline ow_plexil/src/plans/FaultClearDemo.plp:20]: I think the camera trigger message might stand alone as the only way you can ever call a lander operation using a message, aside from publishing an action goal message of course, but that's really just calling an action without a client.

**Review by AstroStucky — APPROVED** [review]
Test works and code looks good!

File: ow_plexil/src/plans/FaultClearDemo.plp (hunk 2 above)
Comment (kmdalal) [reply ow_plexil/src/plans/FaultClearDemo.plp:20]: Pan/Tilt can also be commanded with just messages.

**Pull request merged**